### PR TITLE
[AdminService] Support thread dump.

### PR DESCRIPTION
**Pull request by grao1991**
### Description

Add `/threadz` endpoint to dump stack trace for all threads.
Default libunwind doesn't work well due to a compatibility issue with lld, using libdw instead.

### CODE REVIEW

File: aptos-node/Cargo.toml [hunk 1]
```diff
@@ -79,6 +79,7 @@ url = { workspace = true }
 
 [target.'cfg(unix)'.dependencies]
 jemallocator = { workspace = true }
+rstack-self = { workspace = true }
```
Comment (sitalkedia) [inline aptos-node/Cargo.toml:82]: How about using the `backtrace` crate - the API looks much simpler to me - https://crates.io/crates/backtrace

Comment (grao1991) [reply aptos-node/Cargo.toml:82]: That's for a different purpose, we need dump for every threads here.

Comment (sitalkedia) [reply aptos-node/Cargo.toml:82]: It support dumping for every threads no?

Comment (grao1991) [reply aptos-node/Cargo.toml:82]: no

**Comment by grao1991** [comment]
> nit: consider rename to "profiling service"? admin service sounds too centralized?

I intentionally use a broad term. It will include more functionalities than profiling.

File: crates/aptos-admin-service/src/server/thread_dump.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            info!("Failed to dump threads: {e:?}");
+            Ok(reply_with_status(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                e.to_string(),
+            ))
+        },
+    }
+}
+
+async fn start_thread_dump(
```
Comment (msmouse) [inline crates/aptos-admin-service/src/server/thread_dump.rs:64]: weird to be called `start_` while it does finish the whole thing.

**Review by msmouse — APPROVED** [review]

**Pull request merged**